src/controller/java/src/chip/devicecontroller/model/InvokeElement.java

```diff
@@ -45,8 +72,20 @@ public ChipPathId getCommandId() {
     return commandId;
   }
 
+  public boolean hasTlv() {
```

Comment:
Rather than splitting these into "has" and "get" methods (the latter of which will throw if the "has" is false) could these just return an optional?

```
public Optional<byte[]> getTlv()
```

Then the equivalent "has" check is just `element.getTlv().isPresent()`

Same comment for the JSON. To keep the copy on exit for the TLV data you can just do:

```
if (data != null) {
  return Optional.of(data.clone());
} else {
  return Optional.empty();
}
```

Reply:
updated with nullable, thanks